contribs/gmf/src/editing/Snapping.js

```diff
@@ -410,21 +410,39 @@ EditingSnappingService.prototype.getWFSConfig_ = function(treeCtrl) {
 
 /**
  * @param {import("ngeo/layertree/Controller.js").LayertreeController} treeCtrl The layer tree controller
- * @param {string|undefined} newVal New state value
+ * @return {boolean} True if state is on and snapping enabled for that layer.
+ * @private
+ */
+EditingSnappingService.prototype.isSnappingActiveForTreeCtrl_ = function(treeCtrl) {
+  // Note: a snappable treeCtrl can only be a leaf, therefore the only possible
+  //       states are: 'on' and 'off'.
+  if (treeCtrl.getState() !== 'on') {
+    return false;
+  }
+  if (treeCtrl.properties.snapping === undefined) {
+    const node = /** @type {import('gmf/themes.js').GmfLayer} */ (treeCtrl.node);
+    const config = getSnappingConfig(node);
+    return config !== null && config.activated;
+  }
+  const snappingActive = treeCtrl.properties.snappingActive;
+  if (typeof snappingActive !== 'boolean') {
+    throw new Error('Wrong snappingActive type');
+  }
+  return /** @type {boolean} */ (snappingActive);
```

Comment:
```suggestion
  return snappingActive;
```
typescypt typecheck (tsc) understand the previous if/throw and adapt the type :-)

Reply:
:astonished: Thanks for the tip
Fixed